### ISSUE TITLE
Investigate LLD Build Time Discrepancy

### ISSUE THREAD

**Issue by theduke** [bjorn3/rustc_codegen_cranelift#1147]
As discussed on Zulip, build times with `cg_clif` appear to be consistently faster than llvm with `ld`, but slower with `lld`.

https://github.com/theduke/rustc-cranelift-bench/blob/main/timings.csv

**Comment by bjorn3** [comment]
I can reproduce this. While trying to repro I even found a rustc ICE: https://github.com/rust-lang/rust/issues/83138

**Comment by bjorn3** [comment]
Excepts from `-Ztime-passes`:

cg_llvm:

```
time:   0.056; rss:   67MB ->  126MB (  +59MB)  configure_and_expand
time:   0.208; rss:  142MB ->  143MB (   +0MB)  link
time:   0.360; rss:   51MB ->  141MB (  +90MB)  total
```

cg_clif:

```
time:   0.201; rss:   51MB ->  276MB ( +224MB)  configure_and_expand
time:   0.149; rss:  154MB ->  150MB (   -4MB)  link
time:   0.455; rss:   45MB ->  101MB (  +56MB)  total
```

I think this is #927.

**Comment by bjorn3** [comment]
With #927 implemented in b1d14ca05d6b34423c31fd237fdf2d6cd6f70333, I get:

```
time:   0.104; rss:   50MB ->  130MB (  +81MB)  configure_and_expand
time:   0.155; rss:  135MB ->  132MB (   -4MB)  link
time:   0.379; rss:   44MB ->   94MB (  +51MB)  total
```

According to hyperfine cg_clif is now 8% slower than cg_llvm instead of the ~25% I believe it was before.

**Comment by bjorn3** [comment]
I think the difference is small enough now.
